src/datetime.rs

```diff
@@ -218,7 +218,10 @@ impl FromStr for Datetime {
                     return Err(DatetimeParseError { _private: () })
                 }
                 chars = whole[end..].chars();
-                match format!("0.{}", &whole[..end]).parse() {
+
+                // truncate to picoseconds precision
+                let last = if end > 12 { 12 } else { end };
```

Comment:
I decided to go with picosecond precision (12 digits), because it's a nice round number and there was already a test that required 10 digits of precision. If you think we need more (or less) precision, I'm happy to tweak this.